ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerMatrixNearFracture-impl.h

```diff
@@ -163,30 +209,36 @@ assembleWithJacobian(
         auto& C = ip_data._C;
         auto& material_state_variables = *ip_data._material_state_variables;
 
-        eps.noalias() = B * nodal_u;
+        eps.noalias() = B * nodal_total_u;
 
         if (!ip_data._solid_material.computeConstitutiveRelation(
                 t, x_position, _process_data.dt, eps_prev, eps, sigma_prev,
                 sigma, C, material_state_variables))
             OGS_FATAL("Computation of local constitutive relation failed.");
 
-        // r_ru = B^T Stress = B^T C B (u + phi*[u])
-        localRhs_ru.noalias() -= B.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
-
-        // r_[u] = (phi*B)^T Stress = (phi*B)^T C B (u + phi*[u])
-        localRhs_du.noalias() -= levelsets * B.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
+        // r_u = B^T * Sigma = B^T * C * B * (u+phi*[u])
+        // r_[u] = (phi*B)^T * Sigma = (phi*B)^T * C * B * (u+phi*[u])
+        local_b_u.noalias() -= B.transpose() * sigma * ip_factor;
+        for (unsigned i=0; i<n_fractures; i++)
+            vec_local_b_g[i].noalias() -= levelsets[i] * B.transpose() * sigma * ip_factor;
 
         // J_uu += B^T * C * B
-        localA_uu.noalias() += B.transpose() * C * B * detJ * wp.getWeight() * integralMeasure;
-
-        // J_u[u] += B^T * C * B * levelset
-        localA_ug.noalias() += B.transpose() * C * levelsets * B * detJ * wp.getWeight() * integralMeasure;
-
-        // J_[u]u += (levelset B)^T * C * B
-        localA_gu.noalias() += levelsets * B.transpose() * C * B * detJ * wp.getWeight() * integralMeasure;
-
-        // J_[u][u] += (levelset B)^T * C * (levelset B)
-        localA_gg.noalias() += levelsets * B.transpose() * C * levelsets * B * detJ * wp.getWeight() * integralMeasure;
+        local_J_uu.noalias() += B.transpose() * C * B * ip_factor;
```

Comment:
yes you are right. but I will not include the optimisation at this stage. Because I'll not maintain the LIE stuff and someone has to take over this, I think readability is more important.